aristanetworks/ansible-avd
# Feat(eos_designs): Shutdown underlay links if the peer device is not deployed

## Change Summary

The host variable `is_deployed: false` setting excludes a device from being configured in CloudVision. This is intended for when a device is not present in the network. However, as expected, if the device is not present, the ports facing this device on a peer will end up being configured but will be down. Not only is this a security concern but it causes the validate role to raise an error as the port is expected to be up when using `eos_config_deploy_cvp`.

This enhancement causes the underlay peer port to be shutdown in `eos_designs` if a new fabric level variable `is_deployed_shutdown_peer_interfaces: true`. This seperate variable will avoid any breaking changes in `eos_designs`.

## Related Issue(s)

Fixes #1744 

## Component(s) name

`arista.avd.eos_designs`

## Proposed changes
Consider the simple topology below:

DC1_SPINE1 DC1_BL2

Inventory file:

```
  DC1_BL2:
    hosts:
      DC1-BL2A:
        # Device configuration is generated by AVD
        # Device is not configured on Cloudvision (configlet is uploaded)
        is_deployed: false
```

fabric variable:

```
`is_deployed_shutdown_peer_interfaces: true`
```

Expected state on DC1_SPINE1:

```
interface Ethernet1
  shutdown
```

## How to test
Set device variable to `is_deployed: false` and fabric variable `is_deployed_shutdown_peer_interfaces: true` and check if the peer port is configured to be shutdown.

Check the debug vars that `is_deployed = false` under the facts:

```
ansible_facts:
  switch:
...
    is_deployed: false
```

And that the peer config is shutting down the port:

```
ethernet_interfaces:
  Ethernet1:
    peer: DC1-LEAF1AA
    peer_interface: Ethernet1
    peer_type: l3leaf
    description: P2P_LINK_TO_DC1-LEAF1AA_Ethernet1
    mtu: 1500
    type: routed
    shutdown: true
    ip_address: 172.31.255.0/31
```

## Checklist

### User Checklist


- N/A

### Repository Checklist
